File: easy/onboarding/JavaScript/onboarding.js
/**
 * CodinGame planet is being attacked by slimy insectoid aliens.
 * <---
 * Hint: To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
 **/

// game loop
while (true) {
    const enemy1 = readline(); // name of enemy 1
    const dist1 = parseInt(readline()); // distance to enemy 1
    const enemy2 = readline(); // name of enemy 2
    const dist2 = parseInt(readline()); // distance to enemy 2

    // Determine which enemy is closer and print its name
    console.log(dist1 < dist2 ? enemy1 : enemy2);
}

File: easy/onboarding/Scala/onboarding.scala
import scala.io.StdIn._

/**
 * CodinGame planet is being attacked by slimy insectoid aliens.
 * <---
 * Hint: To protect the planet, you can implement the pseudo-code provided in the statement, below the player.
 **/
object Player extends App {

    // game loop
    while(true) {
        val enemy1 = readLine // name of enemy 1
        val dist1 = readLine.toInt // distance to enemy 1
        val enemy2 = readLine // name of enemy 2
        val dist2 = readLine.toInt // distance to enemy 2
        
        // Write an action using println
        // To debug: Console.err.println("Debug messages...")
        
        // Determine which enemy is closer and print its name
        println(if (dist1 < dist2) enemy1 else enemy2)
    }
}